endpoints/apiconfig.go

```diff
@@ -769,19 +769,78 @@ func validateField(v reflect.Value, t reflect.StructField) error {
 	if err != nil {
 		return fmt.Errorf("parse tag: %v", err)
 	}
-	if v.Interface() != reflect.Zero(v.Type()).Interface() {
-		return nil
+
+	isZero := v.Interface() == reflect.Zero(v.Type()).Interface()
+	if isZero && tag.required {
```

Comment:
This is the same problem we have with protobufs afaik.
Required fields can't have the zero type of the type.